File: 프로그래머스/unrated/150369. 택배 배달과 수거하기/택배 배달과 수거하기.py
# 그리디 같음
# 최대한 멀리서부터, 최대한 꽉 채워서
def solution(cap, n, deliveries, pickups):
    answer = 0
    d = 0
    p = 0
    for i in range(-1, -n-1, -1):
        d += deliveries[i]
        p += pickups[i]
        while d > 0 or p > 0:
            answer += (n+i+1)*2
            d -= cap
            p -= cap
    return answer

File: 프로그래머스/2/150369. 택배 배달과 수거하기/택배 배달과 수거하기.js
function solution(cap, n, deliveries, pickups) {
    let answer = 0;
    let [d, p] = [0, 0];
    for (let i of Array.from(Array(n), (_, index) => n-index-1)) {
        [d, p] = [d+deliveries[i], p+pickups[i]]
        while (d > 0 || p > 0) {
            [d, p] = [d-cap, p-cap]
            answer += (i+1)*2
        }
    }
    return answer;
}